python/ray/experimental/collective/collective_tensor_transport.py

```diff
@@ -0,0 +1,127 @@
+        # of a tensor in the GPU object store. This function returns an ObjectRef that
+        # points to the tensor metadata.
+        # NOTE(swang): We put this task on the background thread to avoid tasks
+        # executing on the main thread blocking this task.
+
+        return src_actor.__ray_call__.options(concurrency_group="_ray_system").remote(
+            __ray_get_tensor_transport_metadata__, obj_id, tensor_transport
+        )
+
+    @staticmethod
+    def get_collective_metadata(
+        src_actor: "ray.actor.ActorHandle",
+        dst_actor: "ray.actor.ActorHandle",
+        tensor_transport_metadata: TensorTransportMetadata,
+        backend: Optional[str] = None,
+    ) -> TensorTransportMetadata:
+        """
+        Update the communicator name, src/dst rank before sending the GPU object.
+        """
+        from ray.experimental.collective import get_collective_groups
+
+        def __ray_update_collective_metadata__(
+            self: "ray.actor.ActorHandle",
+            tensor_transport_metadata: TensorTransportMetadata,
+            communicator_name: str,
+            src_rank: int,
+            dst_rank: int,
+        ):
+            tensor_transport_metadata.communicator_name = communicator_name
+            tensor_transport_metadata.src_rank = src_rank
+            tensor_transport_metadata.dst_rank = dst_rank
+            return tensor_transport_metadata
+
+        communicators = get_collective_groups(
+            [src_actor, dst_actor],
+            backend=backend,
+        )
+        # TODO(kevin85421): Support multiple communicators.
+        if len(communicators) == 0:
+            raise ValueError(
+                f"No communicators found for actors {src_actor} and {dst_actor}. "
+                "Create a communicator with "
+                "`ray.experimental.collective.create_collective_group` "
+                "before calling actor tasks."
+            )
+        elif len(communicators) > 1:
+            raise ValueError(
+                f"There are {len(communicators)} possible communicators that contain actors {src_actor} and {dst_actor}. "
+                "Currently, GPU objects only support one communicator. Please make sure only "
+                "one communicator exists."
+            )
+        communicator = communicators[0]
+        src_rank = communicator.get_rank(src_actor)
+        if src_rank == -1:
+            raise ValueError(
+                f"Sender actor {src_actor} not found in communicator. "
+                "Please make sure the sender and receiver are in the same communicator."
+            )
+        dst_rank = communicator.get_rank(dst_actor)
+        if dst_rank == -1:
+            raise ValueError(
+                f"Receiver actor {dst_actor} not found in communicator. "
+                "Please make sure the sender and receiver are in the same communicator."
+            )
+        return src_actor.__ray_call__.remote(
```

Comment:
Do we need to call this as a remote task? It seems like we could just update the fields directly at the caller process.